doc: changelogg update for mcuboot upmerges. (#2813)
nrfconnect/sdk-nrf: doc/nrf/releases/release-notes-latest.rst

See the `master branch section in the changelog`_ for a list of the most important changes.

sdk-mcuboot
===========

> The MCUboot fork in |NCS| contains all commits from the upstream MCUboot repository up to and including ``???``, plus some |NCS| specific additions.

Reviewer: What commit?

The following list summarizes the most important changes inherited from upstream MCUboot:

>   * Fixed an issue where a left-over image trailer was sometimes not erased after reset.

Reviewer: Did I interpret the input correctly?

Author: @ru-fu This fixes an issue where an image might be erased, but a trailer left behind.

Reviewer: ```suggestion
  * Fixed an issue where after erasing an image, an image trailer might be left behind.
```

  * Added a ``CONFIG_BOOT_INTR_VEC_RELOC`` option to relocate interrupts to application.
  * Fixed single image compilation with serial recovery.
  * Added support for single-image applications (see :option:`CONFIG_SINGLE_IMAGE_DFU`).
  * Added a ``CONFIG_BOOT_SIGNATURE_TYPE_NONE`` option to disable the cryptographic check of the image.

>   * Reduced the minimum number of members in packet for serial recovery.

Reviewer: What packet are we talking about? Is that clear enough?

Author: smp packet

  * Introduced direct execute-in-place (XIP) mode (see ``CONFIG_BOOT_DIRECT_XIP``).
  * Fixed kramdown CVE-2020-14001.

>   * Added support for pointing to a Zephyr project related key via Kconfig (use private key).

Reviewer: What does "use private key" mean here? Is that the Kconfig option that must be selected?

Author: I mean key which is somewhere in a user project directory. @MarekPieta how to describe this better?

Reviewer 2: Maybe:
`Modified build system to let application use private key path that begins in user project configuration directory`

Author: @ru-fu ^^ ?

Reviewer: ```suggestion
  * Modified the build system to let the application use a private key that is located in the user project configuration directory.
```

  * Removed all nRF-specific aliases to particular hardware peripherals, because they are no longer needed now that nodes can be addressed by node labels.
    For example, you should now use ``DT_NODELABEL(i2c0)`` instead of ``DT_ALIAS(i2c_0)``.

> * Mcuboot:

Reviewer 3: ```suggestion
* MCUboot:
```

  * Fix erase of possible trailer leftover after reset

>   * Added option to relocate interrupts to application

Reviewer 3: Can you give this option's name, to make it easier for users to find?

  * Fix single image compilation with serial recovery
  * Allow single image applications

>   * Allow disable cryptographic check of the image

Reviewer 3: Any references to documentation or Kconfig options?

  * serial_recovery: Reduce minimum number of members in packet

>   * Introduced direct execute-in-place (XIP) mode

Reviewer 3: Same here, reference?

  * Fix kramdown CVE-2020-14001
  * Allow to point zephyr project related key via Kconfig (Use private key)
  * boot: Support for nrf52840 with ecc keys and cryptocell
  * Allow to set VTOR to relay vector before chain-load the application

>   * Allow larger primary slot in swap-move

Reviewer 3: Any references?

Author: This was not patch to the move algo, but to before check. Added more description on that.
https://github.com/JuulLabs-OSS/mcuboot/pull/760
